sources/polyfills/MapLike.ts

```diff
@@ -0,0 +1,64 @@
+declare global {
+	class Map<K, V> {
+		// es2015.collection.d.ts
+		clear(): void;
+		delete(key: K): boolean;
+		forEach(
+			callbackfn: (value: V, key: K, map: Map<K, V>) => void,
+			thisArg?: any,
+		): void;
+		get(key: K): V | undefined;
+		has(key: K): boolean;
+		set(key: K, value: V): this;
+		readonly size: number;
+	}
+}
+
+class MapLike<K, V> {
```

Comment:
i added this to match the `SetLike` class. i made it really barebones since only the `set`, `get`, and `has` methods are being used